ethers-contract/src/event.rs

```diff
@@ -111,13 +111,13 @@ impl<M, D: EthLogDecode> Event<'_, M, D> {
     }
 }
 
-impl<'a, M, D> Event<'a, M, D>
+impl<M, D> Event<M, D>
 where
     M: Middleware,
     D: EthLogDecode,
 {
     /// Returns a stream for the event
-    pub async fn stream(
+    pub async fn stream<'a>(
```

Comment:
perhaps we can do the same for `SubscriptionStream`?

Reply:
This is something I wanted too, but the changes were too many. I will give another try.

Reply:
Ok so, to make `SubscriptionStream` an owned struct, the `Middleware` need some radical changes.

This is because `Middleware::Inner` (or `Middleware::Provider`) must be an `Arc`, otherwise we can not construct `SubscriptionStream` by calling `subscribe` method.

Reply:
I prefer not to do it based on my comment above.